content/gotraining.slide

```diff
@@ -577,12 +577,23 @@ Use `panic` rarely and judiciously in your own code!
 
 - [[https://goplay.space/#h1kA3qKaO6B][Errors]]
 - [[https://goplay.space/#9vu6iJHXMHC][Panic]]
-- [[][Defer and Recover]]
+- [[https://goplay.space/#pMbrba6y9cY][Defer]]
```

Comment:
Wow. that defer sample is perfect. Well done!

In all seriousness, I'd also use your other one too. This sample visually shows how defer works. Your copyFile one is more realistic and attaches the defer resource release to the resource acquisition, which is the most common pattern.

https://goplay.space/#pc68j1BPMxn

Reply:
Done.